draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +428,89 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+# Example Extensions
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
+Extensions can define new semantics for the payload of HTTP Datagrams. The
+extension can then have an endpoint pick an available context ID using the
+allocation service ({{context-id}}) and register that context ID with their
+peer. Registration is the action by which an endpoint informs its peer of the
+semantics and format of a given context ID.
 
 
---- back
+## Registering Contexts with Headers
+
+Extensions can define a new HTTP header to register a context ID with the peer
+endpoint.
+
+As an example, let's use an extension that conveys the time at which a UDP
+packet was received. The extension would first define the format of its HTTP
+Datagram Context Payload field:
+
+~~~
+Context Payload for UDP with Timestamp {
+  Timestamp (64),
+  UDP Payload (..),
+}
+~~~
+{: #ex-dgram title="Example: Format of UDP Payload with Timestamp"}
+
+The extension would also define a new HTTP header (UDP-Timestamps) that includes
```

Comment:
It might pay to prefix this with "Example-" in order to (maybe) avoid a potential misunderstanding about whether this is a real thing.

Reply:
Agreed, done.